### PR TITLE
Run unit tests with pytest on Python >= 3.10

### PR DESCRIPTION
**Pull request by narrieta**
Our tests are compatible with pytest, so this is a good option to execute unit tests on Python >= 3.10. In fact, pytest is recommended over nose2 for new projects.

We were using 'attr' from the nose module to mark tests that require sudo. I did not see an obvious way to do the same with pytest, so I removed the attributed and moved those tests to a different file. Unit tests should not require sudo; these 3 tests are a rare exception and I do not anticipate many more tests to require it.

Also, pytest executes the entire test run within the same process, as opposed to nose, which creates a process per test file. The tests in test_agent modify the global logger and conf objects, which was causing many tests to fail when using pytest. I added cleanup code.

pytest is supported on Python >= 3.8; nose on <= 3.9. This PR uses nose for <= 3.9, pytest for >= 3.10

### CODE REVIEW

File: tests/common/test_telemetryevent.py [hunk 1]
```diff
@@ -19,23 +19,23 @@
 from tests.lib.tools import AgentTestCase
 
 
-def get_test_event(name="DummyExtension", op="Unknown", is_success=True, duration=0, version="foo", evt_type="", is_internal=False,
```
Comment (narrieta) [inline tests/common/test_telemetryevent.py:22]: nose was identifying this as a test; renamed it and moved it to the TestTelemetryEvent class

File: tests/ga/test_cgroupconfigurator.py [hunk 2]
```diff
@@ -526,112 +523,6 @@ def test_start_extension_command_should_disable_cgroups_and_invoke_the_command_d
 
                         self.assertEqual(len(CGroupsTelemetry._tracked), 0, "No cgroups should have been created")
 
-    @skip_if_predicate_true(is_python_version_26_or_34, "Disabled on Python 2.6 and 3.4 for now. Need to revisit to fix it")
```
Comment (narrieta) [inline tests/ga/test_cgroupconfigurator.py:529]: Moved those to separate file.

File: tests/ga/test_cgroupconfigurator_sudo.py [hunk 3]
```diff
@@ -0,0 +1,157 @@
+# Copyright 2018 Microsoft Corporation
```
Comment (narrieta) [inline tests/ga/test_cgroupconfigurator_sudo.py:1]: Moved from test_cgroupconfigurator.py without changes, other than removing "@attr('requires_sudo')"

File: tests/ga/test_extension.py [hunk 4]
```diff
@@ -63,10 +63,6 @@
 SUCCESS_CODE_FROM_STATUS_FILE = 1
 
 
-def do_not_run_test():
```
Comment (narrieta) [inline tests/ga/test_extension.py:66]: nose was identifying this as a test; not used, so I removed it

**Review by maddieford — APPROVED** [review]

**Review by nagworld9 — APPROVED** [review]
wow, it seems we were lucky our tests are compatible with other frameworks with minimal changes. Maybe we happen to use native Apis instead of nose specific ones.

**Comment by narrieta** [comment]
@nagworld9 nose/pytest are just the runners. We are using the standard Python unitest module and the mock module for the actual tests. From nose we were using only 'attr', to mark tests as "requires sudo". Initially we were getting tons of failures with nose2/pytest, but those turned out to be caused by AgentTest modifying logger/conf. That was a pain to debug, but once solved, most of the errors went away.

**Pull request merged**